webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java

```diff
@@ -50,18 +54,29 @@ public StudentGroupBatch upload(final User user, final MultipartFile file) {
 
         final StudentGroupBatch.Builder studentGroupBatchBuilder = StudentGroupBatch.builder().creator(user.getUsername());
 
-        try {
-            final String digest = DigestUtils.md5Hex(validate(file.getBytes())).toUpperCase();
+        final MessageDigest messageDigest = DigestUtils.getMd5Digest();
+        try (final DigestInputStream digestStream = new DigestInputStream(file.getInputStream(), messageDigest)) {
+            validate(digestStream);
+        } catch (final Exception ex) {
+            studentGroupBatchBuilder
+                    .digest("") //TODO: should digest be optional?
+                    .message(ex.getMessage())
+                    .status(ImportStatus.BAD_DATA);
+            return repository.create(studentGroupBatchBuilder.build());
+        }
+
+        final String digest = Hex.encodeHexString(messageDigest.digest()).toUpperCase();
+        try (final InputStream fileStream = file.getInputStream()) {
             final Properties properties = new Properties();
-            properties.setProperty("content-type", file.getContentType());
+            properties.put(Headers.CONTENT_TYPE, file.getContentType());
+            properties.put(Headers.CONTENT_LENGTH, file.getSize());
```

Comment:
The ArchiveService expects a Long value for the CONTENT_LENGTH.  I just changed this to use the well-known Amazon header keys and set the CONTENT_LENGTH as a Long value.